lib/slack/real_time/concurrency/async.rb

```diff
@@ -15,7 +15,14 @@ class Socket < Slack::RealTime::Socket
           def start_async(client)
             Thread.new do
               ::Async::Reactor.run do
-                client.run_loop
+                ::Async::Reactor.run do
+                  client.run_loop
+                end
+                ::Async::Reactor.run do |task|
+                  ping do |delay|
```

Comment:
Maybe model this as `run_loop` into a `run_ping` that's something that does ping every once in a while and recovers from errors.

Reply:
Can you expand on the types of errors you'd like to recover from as well as what the expected recover steps would be?

Reply:
It can really be anything, but from looking at the code if `send_data` fails in any way the ping worker dies.